specs/agents/metrics.md

```diff
@@ -50,11 +50,109 @@ Agents should not send the `system.process.cgroup.memory.mem.limit.bytes` metric
 
 Agent should record runtime-specific metrics, such as garbage collection pauses. Due to their runtime-specific nature, these will differ for each agent.
 
-When capturing runtime metrics, keep in mind the end use-case: how will they be used? Is the format in which they are recorded appropriate for visualisation in Kibana? Do not record metrics just because it is easy; record them because they are useful.
+When capturing runtime metrics, keep in mind the end use-case: how will they be used? Is the format in which they are recorded appropriate for visualization in Kibana? Do not record metrics just because it is easy; record them because they are useful.
 
 ### Transaction and span breakdown
 
-Agents should record "breakdown metrics", which is a summarisation of how much time is spent per span type/subtype in each transaction group. This is described in detail in the [Breakdown Graphs](https://docs.google.com/document/d/1-_LuC9zhmva0VvLgtI0KcHuLzNztPHbcM0ZdlcPUl64#heading=h.ondan294nbpt) document, so we do not repeat it here.
+Agents should record "breakdown metrics", which is a summarization of how much time is spent per span type/subtype in each transaction group. This is described in detail in the [Breakdown Graphs](https://docs.google.com/document/d/1-_LuC9zhmva0VvLgtI0KcHuLzNztPHbcM0ZdlcPUl64#heading=h.ondan294nbpt) document, so we do not repeat it here.
+
+### Agent Health Metrics
+
+Agents SHOULD expose metrics which allow the analysis of the internal state of the agent, especially including the [reporting/transport mechanism](transport.md).
+
+Many of these metrics are `COUNTER`s: a `COUNTER` is a monotonically increasing metric.
+It is periodically reported by sending the difference of the counter-value now vs its value at the time of the previous report.
+In order to save elasticsearch disk space, zero-increases are not sent.
+
+Example:
+| **Time** | **Counter** | **Reported Metric Value** |
+|----------|-------------|---------------------------|
+| 0s       | 7           | 7                         |
+| 30s      | 10          | 3                         |
+| 1m       | 10          | 0 (not sent)              |
+| 1m30s    | 12          | 2                         |
+
+
+#### Event Count Metrics
+
+Agents send their data to the Intake API of the APM-server in the form of a stream of events (e.g. transactions, spans, metricsets...).
+
+The following metrics counting these events SHOULD be exposed:
```

Comment:
While collecting the raw stats in terms of counts makes sense, I think the dashboard should focus on the percentage of successful vs dropped events. Maybe a stacked bar chart with successful events vs dropped in agent vs dropped in server.

Reply:
Agree. Collecting the raw counts just makes is easier to aggregate them across agents.

Regarding the dropped events in the server: Do we already have a metric for this?

Reply:
Sorry, I was unclear. What I meant is to differ in the same graph between events that the agent dropped due to the queue being full vs events that have been dropped because the APM Server request was unsuccessful.

APM Server does have metrics about how many events did make it successfully into Elasticsearch. But displaying all that in a single graph may be tricky and I don't believe we need to provide that.